### DISCUSSION TITLE
Angular nested components issue using new syntax but works with old.

### DISCUSSION THREAD

**Discussion by TheDevelolper** [storybookjs/storybook#16657] [Help] [answered]
**Describe the bug**
I have a simple component which encapsulates primeNG's card component: 

``` typescript
<p-card [styleClass]="styleClass ?? ''">
  <ng-template pTemplate="header">
    <img *ngIf="logoHeader" src="/assets/images/logo-dark.svg" alt="Cavalcade Software logo" class="logo">
  </ng-template>
  <ng-content></ng-content>
</p-card>
```

The component is exported via it's own module: 
``` typescript
import { NgModule } from '@angular/core';
import { CardComponent } from './card.component';
import { UxSharedModule } from '../shared/ux-shared.module';
@NgModule({
  declarations: [
    CardComponent
  ],
  imports: [
    UxSharedModule
  ],
  exports: [
    CardComponent
  ]
})
export class CardModule { }
```

Then my story looks like this: 

``` typescript
export default {
  title: "Card component",
  moduleMetadata: moduleMetadata({
    imports: [
      UxSharedModule,
      CardModule
    ],

    declarations: [CardComponent, Card]
  }),
  component: CardComponent,
  template: `
  <c-card>
    Any content you want here!
  </c-card>`,
}
```
When I initially load storybook everything is fine: 

![image](https://user-images.githubusercontent.com/4026553/141096387-d2c619a3-0cfc-4d97-9b40-c3ccd48b6bfb.png)

however, if I navigate away from the component (to another component) then back again: 

![image](https://user-images.githubusercontent.com/4026553/141096463-7f9139eb-b713-4290-9fdb-87b9487e90b7.png)

and I get the following console error: 

```
NG0304: 'p-card' is not a known element:
1. If 'p-card' is an Angular component, then verify that it is part of this module.
2. If 'p-card' is a Web Component then add 'CUSTOM_ELEMENTS_SCHEMA' to the '@NgModule.schemas' of this component to suppress this message.
```

So it initially works but then it doesn't as I navigate through the storybook. 

If I use the depricated syntax it works just fine: 

``` typescript
storiesOf("Card Component", module)
  .addDecorator(
    moduleMetadata({
      imports: [CardModule],
    })
  ).add('Basic Card', () => ({
    template: `
      <c-card>
        Any content you want here!
      </c-card>`,
    component: CardComponent,
    props: {},
  })).add('Width at 400px', () => ({
    template: `
    <div style="max-width: 400px">
      <c-card>
        Any content you want here!
      </c-card>
    </div>`,
    component: CardComponent,
    props: {}
  }), {

  })
```

**System**
Environment Info:

  System:
    OS: Windows 10 10.0.19042
    CPU: (12) x64 Intel(R) Core(TM) i7-8700K CPU @ 3.70GHz 
  Binaries:
    Node: 14.17.0 - C:\Program Files\nodejs\node.EXE       
    Yarn: 1.22.5 - C:\Program Files (x86)\Yarn\bin\yarn.CMD
    npm: 6.14.13 - C:\Program Files\nodejs\npm.CMD
  Browsers:
    Edge: Spartan (44.19041.1266.0), Chromium (95.0.1020.44)
  npmPackages:
    @storybook/addon-docs: ^6.4.0-beta.31 => 6.4.0-beta.31
    @storybook/addon-essentials: ~6.4.0-beta.31 => 6.4.0-beta.31
    @storybook/angular: ~6.4.0-beta.31 => 6.4.0-beta.31
    @storybook/builder-webpack5: ~6.4.0-beta.31 => 6.4.0-beta.31
    @storybook/manager-webpack5: ~6.4.0-beta.31 => 6.4.0-beta.31
    @storybook/preset-scss: ^1.0.3 => 1.0.3

**Comment by shilman** [comment, marked as answer]
That's not the right syntax. Converting your original stories literally to CSF would give:

```js
export default {
  title: "Card Component",
  decorators: [
    moduleMetadata({
      imports: [CardModule],
    }),
  ],
}
export const BasicCard = () => ({
  template: `
    <c-card>
      Any content you want here!
    </c-card>`,
  component: CardComponent,
  props: {},
})
```

**Reply by TheDevelolper** [reply]
Thank you so much! 💕
